app/models/acl_dao.rb

```diff
@@ -0,0 +1,87 @@
+class AclDao
+  attr_accessor :access_token, :client_id, :base_url
+
+  def initialize(token, client_id, base_url)
+    @access_token = token
+    @client_id = client_id
+    @base_url = base_url
+  end
+
+  def get_role(user_id)
+
+    # Sets whether the user is a specific role type
+    # Note: user could be all three.
+
+    results = search_acls_by_user(user_id, 1)
+    noPages = (results['hits']/2000).ceil
+    items = Array.wrap(results['items'])
+
+    (2..noPages+1).each { |i|
+      results = search_acls_by_user(user_id, i)
+      items += Array.wrap(results['items'])
+    }
+
+
+    roles = Set.new
+    items.each {|item|
+      if item['name'].end_with?"DASHBOARD_DAAC_CURATOR"
+        roles << "daac_curator"
+      end
+      if item['name'].end_with?"DASHBOARD_ARC_CURATOR"
+        roles << "arc_curator"
+      end
+      if item['name'].end_with?"DASHBOARD_ADMIN"
+        roles << "admin"
+      end
+    }
+
+    #
+    # returns the highest level role
+    #
+
+    if (roles.include? "admin")
```

Comment:
1. I thought `arc_curator` was a 'higher' role than `daac_curator`, since the former is a system level permission?
2. these `if` statements are short enough that if you want to keep them separate you should make put them on one line (and remove the parens https://github.com/rubocop-hq/ruby-style-guide#no-parens-around-condition), so `return <role> if <role check>` but...
3. you can still link these together with `elsif` I think the one line `if` statements might be fewer lines so that _might_ be preferable

Reply:
#1 No, arc curators lower, arc curators release records to daacs and daacs are allowed to close records.  Hopefully they won't have multiple ACL roles but if you are part of a daac, you should be given this privilege.      I changed the rest to reflect the new (to me) way of doing the if statements.